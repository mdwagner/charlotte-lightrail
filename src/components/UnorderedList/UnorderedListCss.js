import styled from 'styled-components/native';

export const BulletWrapperView = styled.View`
  flexDirection: row;
`;

export const BulletView = styled.View`
  width: 13px;
`;

export const BulletContentView = styled.View`
  flex: 1;
`;

export const ContentText = styled.Text`
  fontSize: 14px;
  color: ${props => props.theme.primaryTextColor};
`;
