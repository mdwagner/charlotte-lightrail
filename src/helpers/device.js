import { Dimensions } from 'react-native';

// identify iPhone model based on device width
const identifyDevice = (deviceScreenProps) => {
  if (deviceScreenProps.width === 414) {
    return 'iPhone 6+';
  } else if (deviceScreenProps.width === 320) {
    return 'iPhone 5';
  }
  return 'iPhone 6';
};

const deviceScreen = Dimensions.get('window');
const deviceName = identifyDevice(deviceScreen);
let deviceVariableSizes = {};
let defaultCenter = {};
let defaultZoom;
if (deviceName === 'iPhone 6+') {
  deviceVariableSizes = {
    calloutHeight: 300,
    nextTimeFontSize: 40,
    stationLabelFontSize: 36,
    nextTimeWrapPaddingVertical: 4,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.12848262558094,
    longitude: -80.84703007335676
  };
  defaultZoom = 10.35016331854935;
} else if (deviceName === 'iPhone 6') {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 36,
    stationLabelFontSize: 30,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.09018630471958,
    longitude: -80.84307324707783
  };
  defaultZoom = 9.228717656576594;
} else {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 33,
    stationLabelFontSize: 22,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest - '
  };
  defaultCenter = {
    latitude: 35.12642689061146,
    longitude: -80.8549019571087
  };
  defaultZoom = 10.10400032344327;
}

const calloutBoxHeight = deviceVariableSizes.calloutHeight - 25; // excludes triangle
const blueBoxHeight = calloutBoxHeight * 0.45; // 45% of calloutBoxHeight
const blackBoxHeight = calloutBoxHeight * 0.55; // 55% of calloutBoxHeight

const insideColPadding = deviceScreen.width * 0.13;
const circleWrapperHeight = 54;
const circleWidth = 40;

export const deviceProps = {
  calloutBoxHeight,
  blueBoxHeight,
  blackBoxHeight,
  deviceVariableSizes,
  deviceScreen,
  defaultCenter,
  defaultZoom,
  deviceName,
  insideColPadding,
  circleWrapperHeight,
  circleWidth
};
