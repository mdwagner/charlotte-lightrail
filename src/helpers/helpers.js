import { Linking } from 'react-native';
import { mapboxApiKey } from './config';

export const startNavigation = async (mode, latlng) => {
  const { latitude, longitude } = latlng;

  const googleMapsURL = `comgooglemaps://?daddr=${latitude},${longitude}&directionsmode=${mode}`;
  const appleMapsURL = `http://maps.apple.com?daddr=${latitude},${longitude}&dirflg=${mode}`;

  const supported = await Linking.canOpenURL(googleMapsURL);
  if (supported) {
    Linking
      .openURL(googleMapsURL)
      .catch(err => console.error('Either Google Maps is not installed or some other error occurred: ', err));
  } else {
    Linking
      .openURL(appleMapsURL)
      .catch(err => console.error('An error occurred: ', err));
  }
};

export const mapboxDistanceAPI = {
  async getDistance(origin, destinations, mode) {
    const coordinates = destinations;
    coordinates.unshift(origin);

    if (__DEV__) {
      const endpoint = `directions-matrix/v1/mapbox/${mode}`;
      coordinates_string = coordinates.map(coordinate => {
        return coordinate.toString();
      });
      coordinates_query = coordinates_string.join(';');
      const url = `https://api.mapbox.com/${endpoint}/${coordinates_query}?access_token=${mapboxApiKey}`;
      const res = await fetch(url);
      return res.json();
    } else {
      const endpoint = `distances/v1/mapbox/${mode}`;
      const url = `https://api.mapbox.com/${endpoint}?access_token=${mapboxApiKey}`;
      const res = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          coordinates,
        }),
      });
      return res.json();
    }
  },
};