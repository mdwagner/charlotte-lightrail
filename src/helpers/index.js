export { withHelpers, withStaticData } from './hoc';