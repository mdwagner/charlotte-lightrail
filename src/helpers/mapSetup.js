import { blueStops } from './config';
import { getNextTrainTime } from './scheduleCalcs';

export const getAnnotations = () =>
  blueStops.map((stop, index) => ({
    coordinates: [stop.latlng.latitude, stop.latlng.longitude],
    type: 'point',
    title: stop.mapLabel,
    annotationImage: {
      source: {
        uri: 'blueMarker'
      },
      height: 32,
      width: 32
    },
    id: `marker-${index}`
  }));

export const getStopCallouts = () => {
  const stopCallouts = {};
  blueStops.forEach((stop, index) => {
    const nextInbound = getNextTrainTime('inbound', index);
    const nextOutbound = getNextTrainTime('outbound', index);
    stopCallouts[`stopCallout${index}`] = {
      durationText: null,
      inbound: nextInbound,
      outbound: nextOutbound
    };
  });
  return stopCallouts;
};
