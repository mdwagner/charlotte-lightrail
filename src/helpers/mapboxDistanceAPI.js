import { mapboxApiKey } from './config';

export const mapboxDistanceAPI = {
  async getDistance(origin, destinations, mode) {
    const coordinates = destinations;
    coordinates.unshift(origin);

    if (__DEV__) {
      const endpoint = `directions-matrix/v1/mapbox/${mode}`;
      const coordinatesString = coordinates.map(coordinate => coordinate.toString());
      const coordinatesQuery = coordinatesString.join(';');
      const url = `https://api.mapbox.com/${endpoint}/${coordinatesQuery}?access_token=${mapboxApiKey}`;
      const response = await fetch(url);
      return response.json();
    }

    const endpoint = `distances/v1/mapbox/${mode}`;
    const url = `https://api.mapbox.com/${endpoint}?access_token=${mapboxApiKey}`;
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        coordinates
      })
    });
    return response.json();
  }
};
