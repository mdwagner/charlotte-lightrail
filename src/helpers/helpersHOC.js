import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';

export function withHelpers(Component) {
  return class extends React.Component {
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}
