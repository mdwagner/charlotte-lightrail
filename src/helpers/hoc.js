import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';
import staticData from './staticData.json';

export function withHelpers(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}

export function withStaticData(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        staticData
      };
      return (
        <Component {...this.props} staticData={newProps} />
      );
    }
  };
}