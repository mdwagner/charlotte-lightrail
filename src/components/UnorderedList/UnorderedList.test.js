import React from 'react';
import { snapshotStyledComponent } from '../../helpers';
import UnorderedList from './UnorderedList';

describe('<UnorderedList />', () => {
  let props;

  beforeAll(() => {
    props = {
      content: [
        'Find the closest station',
        'Get directions, on foot, or by car',
        'See when the next train is coming',
        'See the full rail schedule',
        'See these FAQs about riding the rails'
      ]
    };
  });

  it('renders correctly', () => {
    snapshotStyledComponent(
      <UnorderedList {...props} />
    );
  });
});
