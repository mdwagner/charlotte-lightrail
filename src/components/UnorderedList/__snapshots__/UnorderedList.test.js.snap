exports[`<UnorderedList /> renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        Find the closest station
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        Get directions, on foot, or by car
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See when the next train is coming
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See the full rail schedule
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See these FAQs about riding the rails
      </Text>
    </View>
  </View>
</View>
`;
